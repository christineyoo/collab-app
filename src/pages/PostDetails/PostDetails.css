.post-details {
  padding: 30% 0 10% 0;
}

#post-details-bg {
  background-image: url('../../images/books.png');
}

@media all and (min-width: 800px) {
  .post-details {
    padding: 5% 0 5% 0;
    font-size: 2rem;
  }
}
