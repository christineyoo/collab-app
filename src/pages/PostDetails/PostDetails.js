import React, { Component } from 'react';
import ApiContext from '../../ApiContext.js';
import Footer from '../../Organisms/Footer/Footer.js';
import Post from '../../Molecules/Post/Post';
import SideBar from '../../Organisms/SideBar/SideBar.js';
import UserNav from '../../Organisms/Nav/UserNav.js';
import './PostDetails.css';

class PostDetails extends Component {
  static contextType = ApiContext;

  displayPost = () => {
    const copyPosts = this.context.posts || [];
    const filteredPost = copyPosts.filter(
      (post) => +post.id === +this.props.match.params.postId
    );
    const formattedFilteredPost = filteredPost.map((post, i) => {
      return (
        <ApiContext.Consumer key={i}>
          {(context) => (
            <Post
              post_id={this.props.match.params.postId}
              post_name={post.post_name}
              content={post.content}
              modified={post.modified}
              author={post.author}
              group_id={post.group_id}
            />
          )}
        </ApiContext.Consumer>
      );
    });
    return formattedFilteredPost;
  };

  render() {
    return (
      <>
        <UserNav />
        <div id='post-details-bg'>
          <header className='post-details'>
            <h1 id='post-details-h1'>Post Details</h1>
            <h2>View a post's full content, delete, edit, or comment on a post. </h2>
          </header>
          <div className='flex-container'>
            <div className='flex-1'>
              <SideBar />
            </div>
            <div className='flex-3'>{this.displayPost()}</div>
          </div>
        </div>

        <Footer />
      </>
    );
  }
}

export default PostDetails;
