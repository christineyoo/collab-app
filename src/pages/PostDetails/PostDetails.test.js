import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import PostDetails from './PostDetails';

it('renders PostDetails UI as expected', () => {
  const tree = renderer
    .create(
      <Router>
        <PostDetails />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
