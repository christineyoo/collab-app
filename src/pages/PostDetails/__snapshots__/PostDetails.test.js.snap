// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PostDetails UI as expected 1`] = `
Array [
  <nav>
    <a
      exact={true}
      href="/"
      id="nav"
      onClick={[Function]}
      style={
        Object {
          "fontFamily": "Abril Fatface, cursive",
        }
      }
    >
      Collab
    </a>
  </nav>,
  <div
    id="post-details-bg"
  >
    <header
      className="post-details"
    >
      <h1
        id="post-details-h1"
      >
        Post Details
      </h1>
      <h2>
        View a post's full content, delete, edit, or comment on a post.
         
      </h2>
    </header>
    <div
      className="flex-container"
    >
      <div
        className="flex-1"
      >
        <a
          aria-current={null}
          href="/all-posts"
          onClick={[Function]}
        >
          <div
            className="group-card"
            id="group-all-posts"
          >
            All Posts
          </div>
        </a>
        <a
          aria-current={null}
          href="/group/1"
          onClick={[Function]}
        >
          <div
            className="group-card"
          >
            Algebra 1
          </div>
        </a>
        <a
          aria-current={null}
          href="/group/2"
          onClick={[Function]}
        >
          <div
            className="group-card"
          >
            Geometry
          </div>
        </a>
        <a
          aria-current={null}
          href="/group/3"
          onClick={[Function]}
        >
          <div
            className="group-card"
          >
            Algebra 2
          </div>
        </a>
        <a
          aria-current={null}
          href="/group/4"
          onClick={[Function]}
        >
          <div
            className="group-card"
          >
            Intro to Data Science
          </div>
        </a>
        <a
          href="/add-post"
          onClick={[Function]}
        >
          <div
            className="group-card"
            id="group-add-post"
          >
             + Add a Post
          </div>
        </a>
      </div>
      <div
        className="flex-3"
      />
    </div>
  </div>,
  <footer
    className="footer-style"
  >
    <div
      className="container"
    >
      <h4
        id="footer-h4-hh"
      >
        Copyright © 2021 Christine Yoo
      </h4>
      <ul
        id="footer-menu"
      >
        <li>
          <i
            className="fab fa-linkedin-in"
          />
          <a
            href="https://www.linkedin.com/in/christine-yoo-cy/"
            rel="noreferrer"
            target="_blank"
          >
             LinkedIn
          </a>
        </li>
        <li>
          <i
            className="fab fa-github"
          />
          <a
            href="https://github.com/christineyoo"
            rel="noreferrer"
            target="_blank"
          >
             GitHub
          </a>
        </li>
      </ul>
    </div>
  </footer>,
]
`;
